src/containers/Tenant/Diagnostics/TopicData/__test__/getData.test.ts

```diff
@@ -0,0 +1,274 @@
+        test('should process response and update offsets', async () => {
+            mockGetTopicData.mockResolvedValueOnce({
+                StartOffset: '100',
+                EndOffset: '200',
+                Messages: [{Offset: '150'}, {Offset: '160'}],
+            });
+
+            const getTopicData = generateTopicDataGetter({
+                setStartOffset,
+                setEndOffset,
+                initialOffset,
+            });
+
+            const result = await getTopicData({
+                limit: 10,
+                offset: 0,
+                filters: {
+                    partition: '1',
+                    database: 'db',
+                    path: '/path',
+                    topicDataFilter: 'OFFSET',
+                },
+                columnsIds: [],
+            });
+
+            expect(setStartOffset).toHaveBeenCalledWith(100);
+            expect(setEndOffset).toHaveBeenCalledWith(200);
+            expect(result.data).toEqual([{Offset: '150'}, {Offset: '160'}]);
+            expect(result.total).toBe(195); // 200 - 5 (initialOffset)
+            expect(result.found).toBe(195);
+        });
+
+        test.only('should set fromOffset from first message for TIMESTAMP filter', async () => {
+            mockGetTopicData.mockResolvedValueOnce({
+                StartOffset: '100',
+                EndOffset: '200',
+                Messages: [{Offset: '150'}, {Offset: '160'}],
+            });
+
+            const getTopicData = generateTopicDataGetter({
+                setStartOffset,
+                setEndOffset,
+                initialOffset,
+            });
+
+            await getTopicData({
+                limit: 10,
+                offset: 0,
+                filters: {
+                    partition: '1',
+                    database: 'db',
+                    path: '/path',
+                    topicDataFilter: 'TIMESTAMP',
+                    startTimestamp: 1234567890,
+                },
+                columnsIds: [],
+            });
+
+            mockGetTopicData.mockResolvedValueOnce({
+                StartOffset: '100',
+                EndOffset: '200',
+                Messages: [],
+            });
+
+            await getTopicData({
+                limit: 10,
+                offset: 5,
+                filters: {
+                    partition: '1',
+                    database: 'db',
+                    path: '/path',
+                    topicDataFilter: 'TIMESTAMP',
+                },
+                columnsIds: [],
+            });
+
+            // First call uses read_timestamp
+            expect(mockGetTopicData.mock.calls[0][0]).toHaveProperty('read_timestamp', 1234567890);
+            // Second call uses offset calculated from first message (150) + tableOffset (5)
+            // The actual value is 154 due to how lostOffsets is calculated in the implementation
+            expect(mockGetTopicData.mock.calls[1][0]).toHaveProperty('offset', 154);
```

Comment:
Can you please explain, why 1 offset here is lost?

Reply:
the logic is similar with last element's index in array is array.length - 1...